src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -32,6 +32,8 @@ public function __construct(array $decoders = array())
 
     /**
      * {@inheritdoc}
+     * 
+     * @throws RuntimeException If no decoder is found.
```

Comment:
>A plain @throws Exception means nothing.

it's not a plain one, it's a Serializer Exception.

>And this does not forbid other exceptions to happen when something goes really wrong (would you then also document things like @throws \TypeError in case you don't respect the typehint to be complete ?).

There is a difference between having an exception because you messed up something, e.g. giving the wrong typehint, and an acceptable exception, i.e. an exception you are warned of and should account for.

For example giving the wrong type which results in a `\TypeError` is obviously your mistake and is not worth documenting. If it's a critical part of your application that should not accept failure you should be extremely defensive anyway, but having a serialization exception for example should be something you should be ready to handle, hence be documented.